### PR TITLE
Bottom navigation links

### LINKED ISSUE #211: Add a next or previous link at the bottom of the page
**Is your feature request related to a problem? Please describe.**

I'm reading through the docs and I noticed there is no `next`  or `previous` link at the bottom of the page. I had to scroll back up to the section on the sidebar where the page I want to move to is. I believe this is bad for user experience.

Is this intentional or something we might want to work on?

**Describe the solution you'd like**
A `link` or `button` with either a text/symbol or both that illustrates previous or next sections to the user.

### PR DESCRIPTION
**Pull request by Easybuoy**
Closes #211
## Description

- Ability to navigate to different sections of documentation at the footer of each page
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/chakra-ui-docs/components/BottomLink.vue [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  components: {
+    CLink,
+    CFlex
+  },
+  data: () => ({
+    prev: '',
+    next: ''
+  }),
+  watch: {
+    $route (to, from) {
```
Comment (codebender828) [inline packages/chakra-ui-docs/components/BottomLink.vue:31]: I think to improve perf, let's watch only the route path.

```suggestion
    '$route.path' (to, from) {
```

File: packages/chakra-ui-docs/utils/index.js [hunk 2]
```diff
@@ -17,3 +17,12 @@ export const removeHyphenFromString = (hyphenatedString) => {
...
   return str.join(' ')
 }
+
+export const findNextAndPrevRoute = (path, routes) => {
+  const currentRouteIndex = routes.map(route => route.path).indexOf(path)
+
+  const nextPageLink = routes[currentRouteIndex + 1]
+  const prevPageLink = routes[currentRouteIndex - 1]
+
+  return { prev: prevPageLink ? prevPageLink.path : '', next: nextPageLink ? nextPageLink.path : '' }
```
Comment (codebender828) [inline packages/chakra-ui-docs/utils/index.js:27]: Nice work here. We should also capitalize the path name. For example, `accordion` -> `Accordion`

**Comment by Easybuoy** [comment]
I'm having an issue with some links jumping about as I stated here @codebender828 

https://github.com/chakra-ui/chakra-ui-vue/issues/211#issuecomment-636315088

File: packages/chakra-ui-docs/utils/index.js (hunk 2 above)
Comment (Easybuoy) [reply packages/chakra-ui-docs/utils/index.js:27]: I'm only using the returned path to navigate to the next route. If I were to change it, that would mean we would have a route like `/Constraint-based-design`.

Just confirming if this is what we want? @codebender828

**Comment by peoray** [comment]
@Easybuoy Just checked out the PR and I noticed the issue of the links jumping around happens in the `Getting Started` and `About Chakra UI` section as the `Components` section seems to work fine

**Comment by Easybuoy** [comment]
> @Easybuoy Just checked out the PR and I noticed the issue of the links jumping around happens in the `Getting Started` and `About Chakra UI` section as the `Components` section seems to work fine

Yes, because the components in the `Components` section are listed alphabetically, but the other sections are not.

File: packages/chakra-ui-docs/utils/index.js (hunk 2 above)
Comment (codebender828) [reply packages/chakra-ui-docs/utils/index.js:27]: Oh I think I might have been unclear. I meant for the text that's rendered in the link.

Comment (codebender828) [reply packages/chakra-ui-docs/utils/index.js:27]: See this image for reference:

Chakra UI Vue Docs links

In order to get actual readable text back, we might need to change the values from kebab-case to sentence case before capitalizing it.

**Comment by codebender828** [comment]
> > @Easybuoy Just checked out the PR and I noticed the issue of the links jumping around happens in the `Getting Started` and `About Chakra UI` section as the `Components` section seems to work fine
> 
> Yes, because the components in the `Components` section are listed alphabetically, but the other sections are not.

I see. That is an issue since the files are co-located, it may require a bit of thought to fix this. It might require that we place then in their own directories.

But to begin with, I think we should probably only show it on the Components section, while we think about how to include it for the other sections.

File: packages/chakra-ui-docs/utils/index.js (hunk 2 above)
Comment (Easybuoy) [reply packages/chakra-ui-docs/utils/index.js:27]: Oh, that makes sense, I just had `Prev` and `Next` in there before, thanks for clearing that up 👍🏽

**Comment by Easybuoy** [comment]
> > > @Easybuoy Just checked out the PR and I noticed the issue of the links jumping around happens in the `Getting Started` and `About Chakra UI` section as the `Components` section seems to work fine
> > 
> > 
> > Yes, because the components in the `Components` section are listed alphabetically, but the other sections are not.
> 
> I see. That is an issue since the files are co-located, it may require a bit of thought to fix this. It might require that we place then in their own directories.
> 
> But to begin with, I think we should probably only show it on the Components section, while we think about how to include it for the other sections.

Okay, I'd implement showing  for only the Components section

**Comment by peoray** [comment]
@Easybuoy Might be a long shot but instead of using the routes dynamically, why not use the array that contains the different route name? Would be open to a better solution.

@codebender828 I noticed a different `topNavLinks` and `aboutNavLinks` was used in the `SideNavContent` component that doesn't tally with the ones in `all-routes.js`. Is this intentional?

**Comment by Easybuoy** [comment]
> @Easybuoy Might be a long shot but instead of using the routes dynamically, why not use the array that contains the different route name in the `all-routes.js`? file Would be open to a better solution.
> 
> @codebender828 I noticed a different `topNavLinks` and `aboutNavLinks` was used in the `SideNavContent` component that doesn't tally with the ones in `all-routes.js`. Is this intentional?

I tried finding out how to get the section I'm currently on, but I wasn't able to, If I were able to do that, I'd just use the different route setup in the `all-routes.js` file to get the correct pages

**Comment by codebender828** [comment]
> @Easybuoy Might be a long shot but instead of using the routes dynamically, why not use the array that contains the different route name in the `all-routes.js`? file Would be open to a better solution.
> 
> @codebender828 I noticed a different `topNavLinks` and `aboutNavLinks` was used in the `SideNavContent` component that doesn't tally with the ones in `all-routes.js`. Is this intentional?

They should be the same. Sorry about the discrepancies. This site was pretty much built on the fly. So I forgot to clean up some things after it worked.

The `all-routes.js` file should tally with all values elsewhere. The `topNavLinks` array should also be a member array of the `all-routes` file.

We can make them tally so we can get consistency and a single source of truth. That could be a beginning place for the categorization of the routes. What do you guys think? do you have some ideas around this? @Easybuoy @peoray

**Comment by peoray** [comment]
@codebender828 I totally agree, the `all-routes.js` should be the single source of truth. The data in the sidebar should be removed and the component should be adjusted.

**Comment by peoray** [comment]
> > @Easybuoy Might be a long shot but instead of using the routes dynamically, why not use the array that contains the different route name in the `all-routes.js`? file Would be open to a better solution.
> > @codebender828 I noticed a different `topNavLinks` and `aboutNavLinks` was used in the `SideNavContent` component that doesn't tally with the ones in `all-routes.js`. Is this intentional?
> 
> I tried finding out how to get the section I'm currently on, but I wasn't able to, If I were able to do that, I'd just use the different route setup in the `all-routes.js` file to get the correct pages

I have been able to get this to work. It would require some adjustments from your current PR.

**Comment by Easybuoy** [comment]
> > > @Easybuoy Might be a long shot but instead of using the routes dynamically, why not use the array that contains the different route name in the `all-routes.js`? file Would be open to a better solution.
> > > @codebender828 I noticed a different `topNavLinks` and `aboutNavLinks` was used in the `SideNavContent` component that doesn't tally with the ones in `all-routes.js`. Is this intentional?
> > 
> > 
> > I tried finding out how to get the section I'm currently on, but I wasn't able to, If I were able to do that, I'd just use the different route setup in the `all-routes.js` file to get the correct pages
> 
> I have been able to get this to work. It would require some adjustments from your current PR.

Okay, that's fine, what do I need to do

**Comment by peoray** [comment]
> > > > @Easybuoy Might be a long shot but instead of using the routes dynamically, why not use the array that contains the different route name in the `all-routes.js`? file Would be open to a better solution.
> > > > @codebender828 I noticed a different `topNavLinks` and `aboutNavLinks` was used in the `SideNavContent` component that doesn't tally with the ones in `all-routes.js`. Is this intentional?
> > > 
> > > 
> > > I tried finding out how to get the section I'm currently on, but I wasn't able to, If I were able to do that, I'd just use the different route setup in the `all-routes.js` file to get the correct pages
> > 
> > 
> > I have been able to get this to work. It would require some adjustments from your current PR.
> 
> Okay, that's fine, what do I need to do

Sorry about that, refer to the original comment :)

**Comment by Easybuoy** [comment]
> * and `abou`

Great idea 👌🏾

File: packages/chakra-ui-docs/utils/index.js [hunk 3]
```diff
@@ -18,11 +20,33 @@ export const removeHyphenFromString = (hyphenatedString) => {
...
+  const nextPageLink = nextPage
+    ? {
+      ...nextPage,
+      name: (nextPage.name =
+          nextPage.name.charAt(0).toUpperCase() + nextPage.name.slice(1))
+    }
+    : ''
 
-  return { prev: prevPageLink ? prevPageLink.path : '', next: nextPageLink ? nextPageLink.path : '' }
+  return { prev: prevPageLink, next: nextPageLink }
```
Comment (peoray) [inline packages/chakra-ui-docs/utils/index.js:51]: Great work :)

What's the difference between the (`prevPage` and `nextPage`) and (prevPageLink and nextPageLink), seems redundant to me. The reason is cause I commented out the page-link code and use prev-next-page and it works as expected

File: packages/chakra-ui-docs/utils/index.js (hunk 2 above)
Comment (peoray) [reply packages/chakra-ui-docs/utils/index.js]: In your current commit, the text is too close to the bottom of the page. Also, a border around the links might make it stand out more

Comment (peoray) [reply packages/chakra-ui-docs/utils/index.js]: In your current commit, the text is too close to the bottom of the page. Also, a border around the links might make it stand out more

File: packages/chakra-ui-docs/utils/index.js (hunk 3 above)
Comment (Easybuoy) [reply packages/chakra-ui-docs/utils/index.js:51]: Yeah, good point, I did not need it after implementing the new changes, nice catch 👍🏽

**Comment by codebender828** [comment]
I also noticed that I can still focus on the link if it has an empty string. I think if it has no `to` value or is an empty string, we should remove the `href` attribute or not render the anchor.

image

**Comment by Easybuoy** [comment]
Feedback implemented @codebender828

**Comment by codebender828** [comment]
@all-contributors please add @Easybuoy for docs

**Comment by peoray** [comment]
Probably merged, but just figure I should point out that the border is too close to the bottom of the page. i think I mentioned that earlier. Possible, it's from me. My broswer is Firefox, I can provide screenshots if no one is having the same issue

**Pull request merged**